src/include/brain/kd_tree.h

```diff
@@ -0,0 +1,88 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// kd_tree.h
+//
+// Identification: src/include/brain/kd_tree.h
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "common/macros.h"
+#include "annoy/annoylib.h"
+#include "annoy/kissrandom.h"
+#include "brain/cluster.h"
+
+namespace peloton {
+namespace brain {
+
+//===--------------------------------------------------------------------===//
+// KDTree
+//===--------------------------------------------------------------------===//
+
+class KDTree {
+ public:
+  KDTree(int num_features)
+      : num_features_(num_features), index_(num_features) {}
+
+  void Insert(Cluster *cluster) {
+    index_.unbuild();
+    index_.add_item(size_, cluster->GetCentroid().data());
+    index_.build(2 * num_features_);
+    cluster->SetIndex(size_);
+    clusters_[size_] = cluster;
+    size_++;
+  }
+
+  void Update(UNUSED_ATTRIBUTE Cluster *cluster) {
```

Comment:
I understand that the parameter is unused because the tree is static, but we should add comments and TODO for this.